src/sage/graphs/digraph_generators.py

```diff
@@ -758,6 +759,64 @@ def nauty_directg(self, graphs, options='', debug=False):
             if line and line[0] == '&':
                 yield DiGraph(line[1:], format='dig6')
 
+    def nauty_posetg(self, options='', debug=False):
+        r"""
+        Return a generator which creates all posets using ``nauty``.
+
+        Here a poset is seen through its Hasse diagram, which is
+        an acyclic and transitively reduced digraph.
+
+        INPUT:
+
+        - ``options`` -- string (default: ``""``); a string passed to
+          ``genposetg`` as if it was run at a system command line.
+          At a minimum, you *must* pass the number of vertices you desire
+          and a choice between ``o`` and ``t``` for the output order.
+
+        - ``debug`` -- boolean (default: ``False``); if ``True`` the first line
+          of ``genposetg``'s output to standard error is captured and the first
+          call to the generator's ``next()`` function will return this line as a
+          string. A line leading with ">A" indicates a successful initiation of
+          the program with some information on the arguments, while a line
+          beginning with ">E" indicates an error with the input.
+
+        The possible options, obtained as output of ``genposetg --help``::
+
+            n: the number of vertices
+            o: digraph6 output in arbitrary order
+            t: digraph6 output in topological order
+            q: supresses statistics except for the final count
```

Comment:
Statistics are written in stderr and never reported by this function. 
So either, option `q` should always be used and not documented, or you should find a way to report statistics.